kubernetes-client-openapi/src/main/java/io/micronaut/kubernetes/client/openapi/KubernetesHttpClientFactory.java

```diff
@@ -0,0 +1,144 @@
+import io.micronaut.http.client.filter.ClientFilterResolutionContext;
+import io.micronaut.http.client.filter.DefaultHttpClientFilterResolver;
+import io.micronaut.http.client.netty.DefaultHttpClient;
+import io.micronaut.http.client.netty.NettyClientCustomizer;
+import io.micronaut.http.codec.MediaTypeCodecRegistry;
+import io.micronaut.http.netty.body.NettyByteBufMessageBodyHandler;
+import io.micronaut.http.netty.body.NettyCharSequenceBodyWriter;
+import io.micronaut.http.netty.body.NettyJsonHandler;
+import io.micronaut.http.netty.body.NettyJsonStreamHandler;
+import io.micronaut.http.netty.body.NettyWritableBodyWriter;
+import io.micronaut.json.JsonMapper;
+import io.micronaut.json.codec.JsonMediaTypeCodec;
+import io.micronaut.json.codec.JsonStreamMediaTypeCodec;
+import io.micronaut.kubernetes.client.openapi.ssl.KubernetesClientSslBuilder;
+import io.micronaut.kubernetes.client.openapi.ssl.KubernetesPrivateKeyLoader;
+import io.micronaut.kubernetes.client.openapi.config.KubeConfig;
+import io.micronaut.kubernetes.client.openapi.config.KubernetesClientConfiguration;
+import io.micronaut.runtime.ApplicationConfiguration;
+import io.micronaut.websocket.context.WebSocketBeanRegistry;
+import io.netty.channel.Channel;
+import io.netty.channel.MultithreadEventLoopGroup;
+import io.netty.channel.socket.nio.NioDatagramChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.util.concurrent.DefaultThreadFactory;
+import jakarta.inject.Named;
+import jakarta.inject.Singleton;
+
+import java.net.URI;
+import java.util.Collections;
+
+/**
+ * Factory for kubernetes http client.
+ */
+@Factory
+@Context
+@Internal
+@BootstrapContextCompatible
+@Requires(beans = KubernetesClientConfiguration.class)
+class KubernetesHttpClientFactory {
+    static final String CLIENT_ID = "kubernetes-client";
+
+    private final KubeConfig kubeConfig;
+    private final KubernetesPrivateKeyLoader kubernetesPrivateKeyLoader;
+    private final ResourceResolver resourceResolver;
+    private final DefaultHttpClientFilterResolver defaultHttpClientFilterResolver;
+
+    KubernetesHttpClientFactory(KubernetesClientConfiguration kubernetesClientConfiguration,
+                                KubernetesPrivateKeyLoader kubernetesPrivateKeyLoader,
+                                ResourceResolver resourceResolver,
+                                DefaultHttpClientFilterResolver defaultHttpClientFilterResolver) {
+        kubeConfig = kubernetesClientConfiguration.getKubeConfig();
+        this.kubernetesPrivateKeyLoader = kubernetesPrivateKeyLoader;
+        this.resourceResolver = resourceResolver;
+        this.defaultHttpClientFilterResolver = defaultHttpClientFilterResolver;
+    }
+
+    @Singleton
+    @Named(CLIENT_ID)
+    @BootstrapContextCompatible
+    protected DefaultHttpClient getKubernetesHttpClient() {
+        URI uri = URI.create(kubeConfig.getCluster().server());
+
+        return new DefaultHttpClient(LoadBalancer.fixed(uri),
```

Comment:
I will replace this with http client builder once micronaut-core 4.7.0 gets released

Reply:
Can this code now use `HttpClientRegistry.getClient`? Maybe we need to enhance that API to avoid references to these internal APIs? @yawkat WDYT?

Reply:
`HttpClientRegistry.getClient` can't be used since we need a custom implementation of NettyClientSslBuilder (which is KubernetesClientSslBuilder). Once micronaut-core 4.7.0 gets released entire constructor will be replaced with just a few lines where we set KubernetesClientSslBuilder